File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string City { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/models/userParams.ts
import { User } from './user';
/* to simplify passing argument to MembersService, we will pass this obj
*/
export class UserParams {
    city = 'ontario';
    gender: string;
    minAge = 19;
    maxAge = 99;
    role: string;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
    }
}
